addons/website/static/src/js/editor/widget_link.js

```diff
@@ -26,43 +28,77 @@ weWidgets.LinkTools.include({
     },
     /**
      * Allows the URL input to propose existing website pages.
-     *
+     * Allows the URL input to be filled from document upload.
+     * Automatically creates a callable link if selected content is a phone number.
+     * 
      * @override
      */
     start: async function () {
         var def = await this._super.apply(this, arguments);
-        wUtils.autocompleteWithPages(this, this.$('input[name="url"]'));
+        this.$urlInput = this.$('#o_link_dialog_url_input');
+        wUtils.autocompleteWithPages(this, this.$urlInput);
         this._adaptPageAnchor();
         return def;
     },
-
     //--------------------------------------------------------------------------
     // Private
     //--------------------------------------------------------------------------
+    /**
+     * @private
+     */
+     _openDocumentDialog: function () {
+        const restoreCursor = odooEditorLib.preserveCursor(this.options.wysiwyg.odooEditor.document);
+        const restoreSelection = () => {
+            restoreCursor();
+            /*
+             * Selection is always restored as expected but if user has clicked out of the modal,
+             * toolbar container will be set as display: none
+             */
+            $('#o_we_editor_toolbar_container').css('display', '');
```

Comment:
Can't we replace the import of odooEditorLib + use of preserveCursor/restoreCursor + this css line by a .click() on the link to re-enable the link editor in toolbar ?